org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/editor/WRecordIDEditor.java

```diff
@@ -0,0 +1,367 @@
+		} 
+		else if (WEditorPopupMenu.PREFERENCE_EVENT.equals(evt.getContextEvent()))
+		{
+			if (isShowPreference())
+				ValuePreference.start (getComponent(), gridField, getValue());
+		}
+	}
+
+	@Override
+	public void setReadWrite(boolean readWrite) {
+		m_ReadWrite = readWrite;
+		if(editButton != null)
+			editButton.setDisabled(!readWrite);
+		
+	}
+
+	@Override
+	public boolean isReadWrite() {
+		return m_ReadWrite;
+	}
+
+	public  void propertyChange(PropertyChangeEvent evt)
+	{
+		if (GridField.PROPERTY.equals(evt.getPropertyName()))
+		{
+			setValue(evt.getNewValue(), false);
+		}
+	}
+	
+	@Override
+	public void setValue(Object value) {
+		setValue(value, true);
+	}
+	
+	/**
+	 * 	Set Value
+	 *	@param value value
+	 *	@param fire data binding
+	 */
+	
+	private void setValue (Object value, boolean fire) {
+		if (value == null || value.toString().trim().length() == 0)
+		{
+			oldValue = null;
+			value = null;
+			recordTextBox.setValue("");
+		}
+		else {
+			if(oldTableID != tableIDGridField.getValue()) {
+				oldTableID = tableIDGridField.getValue();
+			}
+			if(value != null && oldTableID != null) {
+				Integer recordID = Integer.parseInt(String.valueOf(value));
+				Integer tableID = Integer.parseInt(String.valueOf(oldTableID));
+				if(recordID != null && tableID != null)
+					recordTextBox.setValue(getIdentifierColumns(tableID, recordID));
+			}
+		}
+
+//		Data Binding
+		if (fire) {
+			ValueChangeEvent changeEvent = new ValueChangeEvent(this, this.getColumnName(), oldValue, value);
+			super.fireValueChange(changeEvent);
+		}
+		oldValue = value;
+
+	}
+
+	@Override
+	public Object getValue() {
+		return oldValue;
+	}
+
+	@Override
+	public String getDisplay() {
+		if(((Integer) tableIDGridField.getValue() != null) && ((Integer) gridField.getValue() != null))
+			return getIdentifierColumns((Integer) tableIDGridField.getValue(), (Integer) gridField.getValue());
+		else
+			return "";
+	}
+	
+	@Override
+    public String getDisplayTextForGridView(Object value) {
+		String s = "";
+		if(value != null &&  value.equals(gridField.getValue())) {
+			this.setValue(value);
+			s = getDisplay();
+			if ("<0>".equals(s)) {
+			s = "";
+			}
+		}
+		else {
+			this.setValue(value);
+			s = "<"+value+">";
+		}
+		return s;
+    }
+
+	@Override
+	public void onEvent(Event event) throws Exception {
+		if(event.getName().equalsIgnoreCase(Events.ON_CLICK)) {
+			if(event.getTarget().equals(zoomButton)) {
+				actionZoom();
+			}
+			else if(event.getTarget().equals(editButton)) {
+				new WRecordIDDialog(recordTextBox.getPage(), this, tableIDGridField);
+			}
+		}
+		else if(event.getName().equalsIgnoreCase(Events.ON_RIGHT_CLICK)) {
+			if(event.getTarget().equals(wrapperDiv)) {
+				popupMenu.open(wrapperDiv);
+			}
+		}
+	}
+
+	/**
+	 * Get Identifier String from AD_Table_ID and Record_ID
+	 * @param tableID
+	 * @param recordID
+	 * @return String
+	 */
+	public static String getIdentifierColumns(int tableID, int recordID) {
```

Comment:
if you can get the lookup (like in WRecordIDDialog), it is better to use the display text from lookup instead. Reason being that would make it consistent with other part of the application.